vita3k/mem/src/mem.cpp

```diff
@@ -56,7 +63,7 @@ static void alloc_inner(MemState &state, Address address, size_t page_count, All
     const void *const ret = VirtualAlloc(memory, aligned_size, MEM_COMMIT, PAGE_READWRITE);
     LOG_CRITICAL_IF(!ret, "VirtualAlloc failed: {}", log_hex(GetLastError()));
 #else
-    mprotect(memory, aligned_size, PROT_READ | PROT_WRITE);
+    assert(mprotect(memory, aligned_size + 1, PROT_READ | PROT_WRITE) == 0);
```

Comment:
The mprotect call will get discarded in release builds. Also, what is the + 1 for?

Reply:
You're right! I've been bamboozled by assert again!

As for the + 1... It seems `aligned_size` wasn't enough for some reason after I aligned memory. I would have a bad access somewhere in unicorn or `load_elf`.

The +1 was just a guess... problems are gone. I don't know why it needs more space!

Reply:
> As for the + 1... It seems aligned_size wasn't enough for some reason after I aligned memory. I would have a bad access somewhere in unicorn or load_elf.

Wdym by not enough? The aligned size is something like 4095 or something? Or Unicorn won't work with 4096?

Reply:
I'm not sure, I can check the bad access address. I think I was getting some bad accesses towards the end of the memory block. + 1 probably adds an entire other page to the mprotect call...

I'm not sure what to do... By default aligned_size is aligned, right? Should I add back what I subtracted from the pointer to the size?